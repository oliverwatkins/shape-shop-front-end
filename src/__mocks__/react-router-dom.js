import React from 'react';


// TODO is this really needed?

//override browser router when running jest test. Effectively disabble it.
const reactRouterDom = require("react-router-dom")
reactRouterDom.BrowserRouter = ({children}) => <div>{children}</div>

module.exports = reactRouterDom