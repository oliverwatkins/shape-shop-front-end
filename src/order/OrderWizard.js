import * as React from 'react';
import ProductListStep from "./Product1Step";
import {connect} from "react-redux";
import {selectDrinks, selectMains, selectSelectedDrinks, selectSelectedProducts} from "../selectors";
import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
import Address from "./AddressStep";
import Summary from "./SummaryStep";
import WhichPayment from "./WhichPaymentStep";
import OKStep from "./OKStep";
import type {AppState, Product} from "../AppState";
import Product2Step from "./Product2Step";
import PaymentStep from "./PaymentStep";

//navigation links
export const wizardPages = {
	PRODUCT_LIST : "/order/productlist",
	DRINK_LIST: "/order/drinklist",
	ADDRESS : "/order/address",
	WHICH_PAYMENT : "/order/whichPayment",
	PAYMENT : "/order/payment",
	SUMMARY : "/order/summary",
	OK: "/order/OK"
}

type Props = {
	products: Array<Product>,
	drinks:Array<Product>,
	selectedProducts: Array<Product>,
	selectedDrinks: Array<Product>,
	address: Address,
	deliveryType: string,
	paymentType: string,
	productsError: string,
}

export class OrderWizard extends React.PureComponent<Props> {
	render() {

		// if (this.props.productsError)
		// 	alert("this.props.productsError")

		return (
			<div className={"order-wizard"}>
				{this.props.productsError && <span className={"error"}>{this.props.productsError}</span>}
				<Router>
						<Switch>
							<Route path={wizardPages.PRODUCT_LIST}>
								<ProductListStep productItems={this.props.products}
																 selectedProducts={this.props.selectedProducts}
																 selectedDrinks={this.props.selectedDrinks}/>
							</Route>
							<Route path={wizardPages.DRINK_LIST}>
								<Product2Step drinks={this.props.drinks} selectedProducts={this.props.selectedProducts}
															selectedDrinks={this.props.selectedDrinks}
									/>
							</Route>
							<Route path={wizardPages.ADDRESS}>
								<Address/>
							</Route>
							<Route path={wizardPages.WHICH_PAYMENT}>
								<WhichPayment/>
							</Route>
							<Route path={wizardPages.SUMMARY}>
								<Summary
									products={this.props.products}
								 	selectedProducts={this.props.selectedProducts}
									selectedDrinks={this.props.selectedDrinks}
								 	address={this.props.address}
									deliveryType={this.props.deliveryType}
									paymentType={this.props.paymentType}
								/>
							</Route>
							<Route path={wizardPages.OK}>
								<OKStep/>
							</Route>
							<Route path={wizardPages.PAYMENT}>
								<PaymentStep
									selectedProducts={this.props.selectedProducts}
									selectedDrinks={this.props.selectedDrinks}
								/>
							</Route>
						</Switch>
				</Router>
			</div>
		);
	}
}

const mapStateToProps = (state: AppState) => {
	return {
		products: selectMains(state),
		productsError: state.products.productsError,
		drinks: selectDrinks(state),
		address: state.order && state.order.address,
		selectedProducts: selectSelectedProducts(state),
		selectedDrinks: selectSelectedDrinks(state),
		deliveryType: state.order && state.order.deliveryType,
		paymentType: state.order && state.order.paymentType,
	};
};


export default connect(
	mapStateToProps,
	null,
)(OrderWizard);