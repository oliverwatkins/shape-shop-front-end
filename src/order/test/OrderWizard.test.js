import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import OrderWizard from "../OrderWizard";

import renderer from 'react-test-renderer';

// Enzyme.configure({adapter: new Adapter()});

xdescribe('OrderWizard test', () => {
	let wrapper;
	const mockStore = configureStore();

	const store = mockStore({
		products: {
			items: [
				{
					name: "prod1",
					quantity: 1,
					price: 123,
					description: "asfd",
					type: "main",
					imageFilename: "",
				},
				{
					name: "prod2",
					quantity: 2,
					price: 124,
					description: "fasfdasfd",
					type: "drink",
					imageFilename: "",
				},
				]
		},
		order: {
			paymentType: "cash",
			deliveryType: "pickup",
			address: {
				name: "fasdfas",
				telephone: "1234444",
				street: "asdfasdfasdf",
				postcode: "sfdsd23",
				username: "asdfasdf"
			}
		},
		login: {
			loginToken: "should something be here?",
			role: "asfd",
			loggingIn: false,
		},
		user: Function, //??
	});

	beforeEach(() => {
		wrapper = mount(
			<Provider store={store}>
				<OrderWizard/>,
			</Provider>
		);
	});
	afterEach(() => {
		wrapper.unmount();
	});

	it('XXXX', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('renders correctly', () => {
		const tree = renderer
			.create(<div >Facebook</div>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

});
