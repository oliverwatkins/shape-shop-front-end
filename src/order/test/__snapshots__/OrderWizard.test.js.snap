// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderWizard test XXXX 1`] = `undefined`;

exports[`OrderWizard test renders correctly 1`] = `
<div>
  Facebook
</div>
`;
